ivy/functional/backends/tensorflow/linear_algebra.py

```diff
@@ -587,10 +589,6 @@ def tensordot(
 ) -> Union[tf.Tensor, tf.Variable]:
     # find type to promote to
     dtype = ivy.as_native_dtype(ivy.promote_types(x1.dtype, x2.dtype))
-
-    # type casting to float32 which is acceptable for tf.tensordot
```

Comment:
seeing this comment, just wanted to confirm if you've tested with all dtypes other than `float32`

Reply:
```
for d in ('float32', 'float64', 'int32', 'complex64', 
           'int64', 'bfloat16', 'complex128', 'float16'):
    x1 = tf.constant([1, 2, 3], dtype=d)
    x2 = tf.constant([1, 2, 3], dtype=d)
    print(d, tensordot(x1, x2, axes=0))
    
float32 tf.Tensor(
[[1. 2. 3.]
 [2. 4. 6.]
 [3. 6. 9.]], shape=(3, 3), dtype=float32)
float64 tf.Tensor(
[[1. 2. 3.]
 [2. 4. 6.]
 [3. 6. 9.]], shape=(3, 3), dtype=float64)
int32 tf.Tensor(
[[1 2 3]
 [2 4 6]
 [3 6 9]], shape=(3, 3), dtype=int32)
complex64 tf.Tensor(
[[1.+0.j 2.+0.j 3.+0.j]
 [2.+0.j 4.+0.j 6.+0.j]
 [3.+0.j 6.+0.j 9.+0.j]], shape=(3, 3), dtype=complex64)
int64 tf.Tensor(
[[1 2 3]
 [2 4 6]
 [3 6 9]], shape=(3, 3), dtype=int64)
bfloat16 tf.Tensor(
[[1 2 3]
 [2 4 6]
 [3 6 9]], shape=(3, 3), dtype=bfloat16)
complex128 tf.Tensor(
[[1.+0.j 2.+0.j 3.+0.j]
 [2.+0.j 4.+0.j 6.+0.j]
 [3.+0.j 6.+0.j 9.+0.j]], shape=(3, 3), dtype=complex128)
float16 tf.Tensor(
[[1. 2. 3.]
 [2. 4. 6.]
 [3. 6. 9.]], shape=(3, 3), dtype=float16)

```

Reply:
awesome, thanks for clarifying!